### PR TITLE
Feature/replace

### PR DESCRIPTION
**Pull request by justy989**

### CODE REVIEW

File: ce_config.c [hunk 1]
```diff
@@ -1886,8 +1889,11 @@ bool key_handler(int key, BufferNode* head, void* user_data)
           case 27: // ESC
           {
                if(config_state->input){
+                    if(config_state->input_key == '/' || config_state->input_key == '?'){
+                         config_state->tab_current->view_input_save->cursor = config_state->start_search;
+                         center_view(config_state->tab_current->view_input_save);
```
Comment (robber-m) [inline ce_config.c:1894]: Awesome!

File: ce_config.c [hunk 2]
```diff
@@ -2658,6 +2660,41 @@ bool key_handler(int key, BufferNode* head, void* user_data)
                          command_buffer->readonly = true;
                          command_buffer->modified = false;
                     } break;
+                    case 'R':
+                    {
+                         YankNode* yank = find_yank(config_state, '/');
+                         if(!yank) break;
+                         if(!buffer->line_count) break;
+                         const char* search_str = yank->text;
+                         char* replace_str = ce_dupe_buffer(config_state->view_input->buffer);
```
Comment (robber-m) [inline ce_config.c]: I'm missing something. Where did the input buffer get opened and populated?

File: ce_config.c [hunk 3]
```diff
@@ -2658,6 +2660,41 @@ bool key_handler(int key, BufferNode* head, void* user_data)
...
+                              replace_count++;
+                         }
+                         if(replace_count){
+                              ce_message("replaced %" PRId64 " matches", replace_count);
+                         }else{
+                              ce_message("no matches found to replace");
+                         }
+                         free(replace_str);
+                    } break;
+                    case 1: // Ctrl + a
```
Comment (robber-m) [inline ce_config.c:2695]: This is great too!

File: ce_config.c [hunk 4]
```diff
@@ -2967,6 +3003,41 @@ bool key_handler(int key, BufferNode* head, void* user_data)
 
                config_state->tab_current = new_tab;
           } break;
+          case 'R':
+               if(config_state->input) break;
+               input_start(config_state, "Replace", key);
```
Comment (robber-m) [inline ce_config.c:3014]: Disregard my previous question

Comment (robber-m) [reply ce_config.c:3014]: I'm looking on my phone so it was hard to see that we were in the end input mode case above

File: ce_config.c [hunk 5]
```diff
@@ -2967,6 +3003,41 @@ bool key_handler(int key, BufferNode* head, void* user_data)
...
+               if(search_str[0] &&
+                  ce_find_string(config_state->tab_current->view_input_save->buffer,
+                                 &config_state->start_search, search_str, &match,
+                                 config_state->search_command.direction)){
+                    ce_set_cursor(config_state->tab_current->view_input_save->buffer,
+                                  &config_state->tab_current->view_input_save->cursor, &match);
+                    center_view(config_state->tab_current->view_input_save);
+               }else{
+                    config_state->tab_current->view_input_save->cursor = config_state->start_search;
+               }
```
Comment (robber-m) [inline ce_config.c:3046]: I'm excited for this. I love incsearch

File: ce_config.c [hunk 6]
```diff
@@ -2997,6 +3068,7 @@ void draw_view_statuses(BufferView* view, BufferView* current_view, VimMode vim_
      mvprintw(view->bottom_right.y, view->top_left.x + 1, " %s%s%s%s ",
               view == current_view ? mode_names[vim_mode] : "",
               modified_string(buffer), buffer->filename, readonly_string(buffer));
+     if(view == current_view) printw("%s %d ", keyname(last_key), last_key);
```
Comment (robber-m) [inline ce_config.c:3078]: `#ifndef NDEBUG`

**Review by robber-m — CHANGES_REQUESTED** [review]

File: ce_config.c (hunk 6 above)
Comment (justy989) [reply ce_config.c:3078]: done !

**Review by robber-m — APPROVED** [review]

**Pull request merged**